libflux/flux/src/cffi.rs

```diff
@@ -455,7 +421,10 @@ pub struct StatefulAnalyzer {
 }
 
 impl StatefulAnalyzer {
-    fn analyze(&mut self, ast_pkg: &ast::Package) -> Result<fluxcore::semantic::nodes::Package> {
+    fn analyze(
+        &mut self,
+        ast_pkg: &ast::Package,
+    ) -> SalvageResult<fluxcore::semantic::nodes::Package, Error> {
```

Comment:
Enables `analyze_with` to both return an error and a semantic graph, as `flux_analyze` could